### PR TITLE
Fix package building for Jessie

### PR DESCRIPTION
**Pull request by Alukardd**
Signed-off-by: Alexey Mochkin <alukardd@alukardd.org>

### CODE REVIEW

File: debian/control [hunk 1]
```diff
@@ -16,6 +16,7 @@ Build-Depends:
 	libboost-filesystem-dev,
 	libidn11-dev,
 	blackhole-dev (>= 0.2.2-1),
+	virtualenv,
```
Comment (bioothod) [inline debian/control]: Does this package exist in other debian/ubuntu versions?
It should probably be:
python-virtualenv | virtualenv

**Comment by Alukardd** [comment]
Oh maybe...
In wheezy it broke package manager
```
Depends: python3, python3-virtualenv
Breaks: python-virtualenv (< 1.11.6)
```

**Comment by bioothod** [comment]
Ubuntu Trusty uses 1.11.4 python-virtualenv.
Why do you want this particular version?

**Comment by shindo** [comment]
Please, do not update changelog here.

**Comment by shindo** [comment]
It looks like virtualenv package in Debian Jessie is Python3 virtualenv:
https://packages.debian.org/jessie/virtualenv

And Python2 virtualenv is python-virtualenv:
https://packages.debian.org/ru/jessie/python-virtualenv

So why should we use Python3 virtualenv?

**Comment by Alukardd** [comment]
@shindo, sorry but i can't do it only if  I deal separate branch or repo for you, because I build package from my repo and in Jessie native format package version can't have "-"(minus symbol).

@bioothod, my condition will accept python-virtualenv 1.11.4 in Ubuntu 14.04, first of all because there is no package virtualenv in trusty repo.

**Comment by Alukardd** [comment]
@shindo, because you use ```virtualenv``` from the ```$PATH``` and in Jessie this binary presented only in package named ```virtualenv```.

**Comment by shindo** [comment]
I don't get what is the problem.
You should update `debian/control` only, and do not touch `debian/changelog`.

**Comment by shindo** [comment]
So what's the binary that is provided by python-virtualenv?

**Comment by Alukardd** [comment]
And here ```debian/control```? GBP use version from changelog...

In Debian Wheezy binary ```virtualenv``` was provided by package ```python-virtualenv```, but in Jesssie this is not true.

**Comment by bioothod** [comment]
Elliptics uses only `python-virtualenv` in its deps, and it compiles just fine in jessie

**Comment by shindo** [comment]
About `debian/changelog`:
It creates new version of the package (you should not do this within this pull request, it's subject of completely different pull request).

About virtualenv:
If I get it right `python-virtualenv` in Jessie provides Python2 virtualenv and `virtualenv` provides Python3 virtualenv.
We should not use Python3 virtualenv.

**Comment by Alukardd** [comment]
I don't interested in my name in your changelog and version doesn't mean nothing for me. If you'll fix package building in Jessie, I will be use your repo again. If you can't accept my pull request please fix it yourself.

```virtualenv``` this is only Build-Depends... Further more (from Debian Jessie)
```
# aptitude why virtualenv
i   python-virtualenv Recommends virtualenv
```

**Comment by bioothod** [comment]
`Recommends` option is a flag, that package it recommends may be useful for the user.
But depending on and bringing in `python3` is not really an option.

What is wrong with just `python-virtualenv` in build deps? Apparently someone provides this package since Elliptics builds successfully.

**Comment by shindo** [comment]
It's not about whose name is in changelog. We try to follow different version numbering scheme (X.Y.Z).
Currently there're a lot of changes made since the previous version that should be included in the changelog.

Your contribution is appreciated. Could you please provide us with the following information:
- what does python-virtualenv provide in your system?
- what error happens when you try to build with the current Build-Depends?

**Comment by Alukardd** [comment]
lol, but only test
```
tests/CMakeLists.txt:26:    COMMAND virtualenv -p "${PYTHON_EXECUTABLE}" . &&
```
From Jessie
```
# dpkg -L python-virtualenv 
/.
/usr
/usr/lib
/usr/lib/python2.7
/usr/lib/python2.7/dist-packages
/usr/lib/python2.7/dist-packages/virtualenv-1.11.6.egg-info
/usr/lib/python2.7/dist-packages/virtualenv-1.11.6.egg-info/top_level.txt
/usr/lib/python2.7/dist-packages/virtualenv-1.11.6.egg-info/entry_points.txt
/usr/lib/python2.7/dist-packages/virtualenv-1.11.6.egg-info/dependency_links.txt
/usr/lib/python2.7/dist-packages/virtualenv-1.11.6.egg-info/not-zip-safe
/usr/lib/python2.7/dist-packages/virtualenv-1.11.6.egg-info/PKG-INFO
/usr/lib/python2.7/dist-packages/virtualenv.py
/usr/share
/usr/share/doc
/usr/share/doc/python-virtualenv
/usr/share/doc/python-virtualenv/index.rst
/usr/share/doc/python-virtualenv/changelog.Debian.gz
/usr/share/doc/python-virtualenv/changelog.gz
/usr/share/doc/python-virtualenv/copyright
/usr/share/doc/python-virtualenv/AUTHORS.txt
```
As you can see this packages no longer provide ```/usr/bin/virtualenv```

**Comment by shindo** [comment]
And what does `virtualenv` package provide?

**Comment by Alukardd** [comment]
Only this binary :-)
```
# dpkg -L virtualenv 
/.
/usr
/usr/bin
/usr/bin/virtualenv
/usr/share
/usr/share/man
/usr/share/man/man1
/usr/share/man/man1/virtualenv.1.gz
/usr/share/doc
/usr/share/doc/virtualenv
/usr/share/doc/virtualenv/changelog.Debian.gz
/usr/share/doc/virtualenv/changelog.gz
/usr/share/doc/virtualenv/copyright
```

**Comment by shindo** [comment]
OK, here's the related bug:
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=752467

Ubuntu did the same for Utopic/Vivid/...

So we actually should use `python-virtualenv | virtualenv`. I'm not sure about `(>= 1.11.6)` version restriction. Why is it needed?

**Comment by Alukardd** [comment]
I don't check all Ubuntus versions and all Debians. This restriction needed for Debian Wheezy because it has package named ```virtualenv``` too but version less then 1.11.6. Also it has condition
```
Breaks: python-virtualenv (< 1.11.6)
```

**Comment by shindo** [comment]
Are you sure? I cannot find it for Wheezy:
https://packages.debian.org/search?keywords=virtualenv

Could you please check what repo this package came from (in Wheezy)?

**Comment by Alukardd** [comment]
Oh, sorry. It's my colleagues add ```sid``` repo for Wheezy server :-(

**Comment by shindo** [comment]
Could you please remove this restriction and changelog's update, and squash your commits into one?

**Comment by Alukardd** [comment]
https://github.com/reverbrain/swarm/pull/67

**Comment by shindo** [comment]
Thanks!

BTW, there's no need to create different pull request. You can update your branch locally (`git rebase -i`) and push it with `--force`.

**Comment by Alukardd** [comment]
I should. Because as I said I built package from my master branch. And gbp doesn't accept minus...

**Comment by shindo** [comment]
Oh, I see.

**Pull request closed without merging**